media-libs/libdiscid/patches/libdiscid-0.6.2.patchset

```diff
@@ -0,0 +1,176 @@
+ libdiscid_la_LIBADD =
+ 
+-#if OS_BEOS
+-#libdiscid_la_LIBADD += -lbe -lroot
+-#endif
++if OS_HAIKU
++libdiscid_la_LIBADD += -lbe -lroot
++libdiscid_la_SOURCES += src/toc.c src/unix.c src/disc_haiku.c
++endif
+ if OS_DARWIN
+ libdiscid_la_LDFLAGS += -framework CoreFoundation -framework IOKit
+ libdiscid_la_SOURCES += src/toc.c src/unix.c src/disc_darwin.c
+diff --git a/configure.ac b/configure.ac
+index 0d21232..accef71 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -45,7 +45,7 @@ AC_DEFINE(libdiscid_EXPORTS, [], "the header is used to create the library")
+ 
+ 
+ case "${host_cpu}-${host_os}" in
+-  #*-beos*)    os=beos;;
++  *-haiku*)    os=haiku;;
+   *-mingw*)   os=win32;;
+   *-cygwin*)  os=win32;;
+   *-freebsd*) os=freebsd ;;
+@@ -63,7 +63,7 @@ case "${host_cpu}-${host_os}" in
+ esac
+ 
+ AC_MSG_NOTICE([Using disc read implementation for: $os])
+-#AM_CONDITIONAL([OS_BEOS], [test x$os = xbeos])
++AM_CONDITIONAL([OS_HAIKU], [test x$os = xhaiku])
+ AM_CONDITIONAL([OS_DARWIN], [test x$os = xdarwin])
+ AM_CONDITIONAL([OS_FREEBSD], [test x$os = xfreebsd])
+ AM_CONDITIONAL([OS_GENERIC], [test x$os = xgeneric])
+diff --git a/src/disc_haiku.c b/src/disc_haiku.c
+new file mode 100644
+index 0000000..1b46c04
+--- /dev/null
++++ b/src/disc_haiku.c
+@@ -0,0 +1,121 @@
++/* --------------------------------------------------------------------------
++
++   MusicBrainz -- The Internet music metadatabase
++
++   Copyright (C) 2009 Shunsuke Kuroda
++   Copyright (C) 2006 Matthias Friedrich
++   Copyright (C) 2000 Robert Kaye
++   Copyright (C) 1999 Marc E E van Woerkom
++   
++   This library is free software; you can redistribute it and/or
++   modify it under the terms of the GNU Lesser General Public
++   License as published by the Free Software Foundation; either
++   version 2.1 of the License, or (at your option) any later version.
++   
++   This library is distributed in the hope that it will be useful,
++   but WITHOUT ANY WARRANTY; without even the implied warranty of
++   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
++   Lesser General Public License for more details.
++   
++   You should have received a copy of the GNU Lesser General Public
++   License along with this library; if not, write to the Free Software
++   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
++   02110-1301  USA
++
++--------------------------------------------------------------------------- */
++#include <stdlib.h>
++#include <stdio.h>
++#include <string.h>
++#include <fcntl.h>
++#include <assert.h>
++#include <unistd.h>
++#include <scsi.h>
++#include <sys/types.h>
++#include <sys/stat.h>
++#include <sys/ioctl.h>
++
++
++#include "discid/discid.h"
++#include "discid/discid_private.h"
++#include "unix.h"
++
++#define NUM_CANDIDATES 2
++
++static char *device_candidates[NUM_CANDIDATES] = {"/dev/disk/atapi/0/slave/raw",
++					         "/dev/disk/atapi/1/slave/raw"};
```

Comment:
Would libdiskid be able to work on devices if they are not in this candidates list?
If not, would it be easily feasible to handle any device instead of just those in the candidate list?

Reply:
> Would libdiskid be able to work on devices if they are not in this candidates list?

Yes, but it cannot automatically detect them.

By specifying `device` here: https://github.com/metabrainz/libdiscid/blob/1a39085ae119c71ed6c113c3e5dda47c4aee06fa/src/disc.c#L160, programs can use other devices given by the user.

It is feasible, though, that we can use some Haiku native API (or simply read the directory of "/dev/disk") to completely solve the problem, but an easier solution would be to add most commonly used devices, namely (the list in the patch is incomplete):
- /dev/disk/atapi/0/master/raw
- /dev/disk/atapi/0/slave/raw
- /dev/disk/atapi/1/master/raw
- /dev/disk/atapi/1/slave/raw

What do you think?

Reply:
Since the answer to my first question is yes I am perfectly OK with your current patch :)

Any program willing to access a device which is not in the candidate list can perfectly specify it.
And I guess most users would probably have their device in the candidate list.

So we can merge the PR.

Reply:
OK, thanks. Would you please wait till I update the candidates list (it's not complete)?

Reply:
Hmm, if you would like to add the `/dev/disk/atapi/{0,1}/master/raw` I think it's a good idea. `/dev/disk/atapi/1/master/raw` can be quite common, after all.